articles/implementing-network-address-translation/index.md

```diff
@@ -0,0 +1,205 @@
+
+### Types of NAT.
+Network address translation can be classified into three:
+1. Static Network Translation (Static NAT)
+2. Dynamic Network Address Translation (Dynamic NAT)
+3. Port Address Translation (PAT)
+
+#### Static NAT
+Static NAT creates a one-to-one mapping between private and public addresses.
+Static NAT is usually configured by a network administrator, and this configuration remains constant.
+
+![Static NAT](/engineering-education/implementing-network-address-traslation/static.jpg)
+In the figure above, PC1 and PC2 want to reach PC3, which is in a remote network.
+But because both are configured with private addresses, which are not routable, they can not access PC3 directly.
+To access PC3, a NAT-enabled router is configured with static NAT, which mapped their  private addresses to public addresses one-to-one, thus allowing them to communicate with PC3.
+Static NAT is useful for a device that needs a dedicated address, such as a web server, but requires an equal number of public addresses for Simultaneous users.
+
+ #### Dynamic NAT
+Dynamic NAT, much the same as static, gives a one-to-one mapping among private and public addresses, yet not at all like static NAT dynamic.
+Dynamic NAT makes a pool of public addresses and assigned them to private addresses on a first-come, first-served basis using matching criteria to determine which private addresses ought to be translated.
+![Dynamic NAT](/engineering-education/implementing-network-address-traslation/dynamic.jpg)
+In the figure above, an organization was assigned four public addresses, but the organization has more than four internal devices that require access to the internet.
+To resolve this problem, the network administrator decides to configure dynamic NAT to allow these devices to access the internet.
+If internal devices have been assigned all the available global addresses, any device requesting a public address to reach the internet will have to wait until one is made available.
+
+#### Port Address Translation (PAT).
+Dynamic NAT is use by an organizations to connect their devices, but some organizations need to have most if not all devices reach the internet.
+If their network is large, it requires a huge set of registered public addresses; thus, it completely defeats NAT's goal.
+Dynamic NAT lessens this problem to some degree. However, if a large percentage of internal hosts need access to the internet, Port Address Translation, also called NAT overload, solves this problem.
+To understand how PAT works, it is important to recall how the host uses the Transmission Control Protocol (TCP) and User Datagram Protocol (UDP) and port numbers to transmit data, which can be found here [Understanding TCP/IP Transport Layer Protocols: TCP and UDP](https://www.section.io/engineering-education/understanding-tcp-ip-transport-layer-protocols/).
```

Comment:
When you reference anyother Section article, you need not type out the entire URL, instead type out `/engineering-education/understanding-tcp-ip-transport-layer-protocols/`.

The reason is, when you specify an URL and the reader clicks on it, it takes time to redirect from one page to another by making a DNS search. Since, you reference a section article, you can just specify the articles' path, which makes it easier and faster to load.